### PR TITLE
Bump package versions for release 2018-04-02

### PR DESCRIPTION
**Pull request by abhipsaMisra**
# Description of the changes

Bumping the package versions for release 2018-04-02

### CODE REVIEW

File: e2e/test/TwinE2ETests.cs [hunk 1]
```diff
@@ -750,10 +750,7 @@ static public void ClassCleanup()
                 Assert.IsTrue(serviceTwin.Properties.Reported.Contains(propName1));
                 String value1 = serviceTwin.Properties.Reported[propName1].ToString();
 
-                // TODO: #243 
-                // If service team fixed the Issue #243 the following Assert.AreNotEqual should fail 
-                // and need to be changed to Assert.AreEqual
-                Assert.AreNotEqual(value1, propEmptyValue);
+                Assert.AreEqual(value1, propEmptyValue);
```
Comment (CIPop) [inline e2e/test/TwinE2ETests.cs]: @abhipsaMisra @zolvarga The tests are going to be in an odd state as our service gets upgraded.
My suggestion is to disable the tests against a GitHub issue (maybe the original one) until things get stable. At that point we can simply enable the test validating the new service behavior and close the bug.

File: iothub/device/src/Microsoft.Azure.Devices.Client.csproj [hunk 2]
```diff
@@ -22,7 +22,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <Version>1.7.0</Version>
+    <Version>1.7.1</Version>
```
Comment (CIPop) [inline iothub/device/src/Microsoft.Azure.Devices.Client.csproj]: What changes do we have to IoTHub DeviceClient?

File: versions.csv [hunk 3]
```diff
@@ -1,10 +1,10 @@
 AssemblyPath, Version
-iothub\device\src\Microsoft.Azure.Devices.Client.csproj, 1.7.0
+iothub\device\src\Microsoft.Azure.Devices.Client.csproj, 1.7.1
```
Comment (CIPop) [inline versions.csv]: See above question.

File: iothub/device/src/Microsoft.Azure.Devices.Client.csproj (hunk 2 above)
Comment (abhipsaMisra) [reply iothub/device/src/Microsoft.Azure.Devices.Client.csproj]: The following change was made to the ProductInfo class: Update ProductInfo to use semver-compliant AssemblyInformationalVersion

File: versions.csv (hunk 3 above)
Comment (abhipsaMisra) [reply versions.csv]: same as above

**Pull request merged**